moshi/src/main/java/com/squareup/moshi/JsonReader.java

```diff
@@ -528,6 +532,11 @@ private Options(String[] strings, okio.Options doubleQuoteSuffix) {
       this.doubleQuoteSuffix = doubleQuoteSuffix;
     }
 
+    /** Returns a copy of this {@link Options Option's} strings. */
+    public Set<String> getStrings() {
+      return unmodifiableSet(new LinkedHashSet<>(Arrays.asList(strings.clone())));
```

Comment:
The LinkedHashSet is not needed here. It creates an extra copy. We don't mutate the backing array so there's no need for the copy.

Reply:
This should be a List, not a Set. Though the elements are unique, the ordering is very important because the rest of the API is focused on indexes.

Reply:
Cleaned up in dfcd74f